pkg/pool-weighted/test/ManagedPool.test.ts

```diff
@@ -95,30 +103,158 @@ describe('ManagedPool', function () {
+    context('join swaps', () => {
+      function itPerformsAJoinSwapCorrectly(
+        doJoinSwap: () => Promise<SwapResult>,
+        queryJoinSwap: () => Promise<BigNumber[]>,
+        queryEquivalentJoin: () => Promise<JoinQueryResult>
+      ) {
+        function isEquivalentToARegularJoin() {
+          it("doesn't revert", async () => {
+            await expect(doJoinSwap()).to.not.be.reverted;
+          });
+
+          it('returns the same amount of BPT as the equivalent join', async () => {
+            const joinSwapResult = await queryJoinSwap();
+            const joinResult = await queryEquivalentJoin();
+
+            // BPT is leaving the Vault and so is represented as a negative value.
+            expect(joinSwapResult[0].mul(-1)).to.be.eq(joinResult.bptOut);
+          });
+
+          it('takes the same amount of tokens as the equivalent join', async () => {
+            const joinSwapResult = await queryJoinSwap();
+            const joinResult = await queryEquivalentJoin();
+
+            expect(joinSwapResult[1]).to.be.eq(joinResult.amountsIn[2]);
+          });
+        }
+
+        context('when swaps are disabled', () => {
+          sharedBeforeEach('disable swaps', async () => {
+            await pool.setSwapEnabled(owner, false);
+          });
+
+          it('it reverts', async () => {
+            await expect(doJoinSwap()).to.be.revertedWith('SWAPS_DISABLED');
+          });
+        });
+
+        context('when swaps are enabled', () => {
+          sharedBeforeEach('enable swaps', async () => {
+            await pool.setSwapEnabled(owner, true);
+          });
+
+          context('when LP allowlist is enabled', () => {
+            sharedBeforeEach('enable allowlist', async () => {
+              await pool.setMustAllowlistLPs(owner, true);
+            });
+
+            context('when trader is allowlisted', () => {
+              sharedBeforeEach('allowlist LP', async () => {
+                await pool.addAllowedAddress(owner, other);
+              });
+
+              isEquivalentToARegularJoin();
+            });
+
+            context('when trader is not allowlisted', () => {
+              it('reverts', async () => {
+                await expect(doJoinSwap()).to.be.revertedWith('ADDRESS_NOT_ALLOWLISTED');
+              });
+            });
+          });
+
+          context('when LP allowlist is disabled', () => {
+            sharedBeforeEach('disable allowlist', async () => {
+              await pool.setMustAllowlistLPs(owner, false);
+            });
+
+            isEquivalentToARegularJoin();
+          });
+        });
+      }
+
+      context('given in', () => {
+        itPerformsAJoinSwapCorrectly(
+          () => pool.swapGivenIn({ in: 1, out: BPT_INDEX, amount: fp(0.1), from: other, recipient: other }),
+          () =>
+            pool.vault.queryBatchSwap({
+              kind: SwapKind.GivenIn,
+              assets: [pool.address, poolTokens.first.address],
```

Comment:
The Pool has more tokens than this. I'm slightly surprised the test passes given this, since we do call the vault to ask for token addresses

Reply:
This array is only for the assets which are involved in the swap. It's for the vault to use to translate `assetInIndex` and `assetOutIndex` into addresses so we only need to provide the token we're joining with and BPT.

Reply:
Ahh sorry - I thought this was `joinPool`'s `assets`